engines/account/app/controllers/account/api/v1/users_controller.rb

```diff
@@ -54,22 +73,32 @@ def update
           end
         end
 
+        # DELETE /api/v1/users/:id
+        #
         def destroy
-          user = Account::User.find(params[:id])
           authorize user
 
           render json: Account::User.destroy(user.id)
         end
 
-        def account
-          render json: UserSerializer.new(current_user)
-        end
-
         private
 
         def users_params
           params.permit(:email, :username, :first_name, :last_name, :password, :password_confirmation)
         end
+
+        def load_user
+          @user = ::Account::User.find_by_id(params[:id])
+          render status: :not_found, nothing: true unless @user
+        end
+
+        def load_signup
+          signup_token = request.headers['HTTP_X_KATUMA_SIGNUP_TOKEN']
```

Comment:
> we'll do the same using another custom header.

Buff. In both cases we just need to send a form with email and token fields. And in both cases API endpoind need to check those fields. I think doing this in headers is overengeniering in frontend and backend